navigating-the-single-cell-landscape-a-comprehensive-workshop-on-scrnaseq/day-1/4-psuedobulk.qmd

```diff
@@ -17,208 +17,532 @@ editor_options:
+                colConnectors = 'grey30')
+
 ```
 
+TD: List notable genes, highlight where they are found, what they do, etc
+
+### From RDS object
+
 ```{r}
-res01 <- results(dds, alpha = "0.01")
-summary(res01)
+res <- readRDS('pseudobulk_analysis/results/cd4_annot_pseudobulk_results.rds')
 ```
 
+Volcano plot is often the ultimate goal of differential expression analysis.
+
 ```{r}
-results(dds, contrast = c("condition", "Flu", "Normal"))
+library(EnhancedVolcano)
+EnhancedVolcano(res,
+                lab = rownames(res),
+                x = 'log2FoldChange',
+                y = 'padj',
+                xlim = c(-8, 8),
+                ylim = c(0, 5),
+                title = 'Volcano Plot: CD4 T cells',
+                subtitle = 'Normal < -  - - -  | -  - - -  > Flu',
+                xlab = 'Log2 Fold Change',
+                ylab = '-log10 FDR',
+                pCutoff = 0.05,
+                FCcutoff = 1.5,
+                pointSize = c(ifelse(res$padj < 0.05 & abs(res$log2FoldChange) > 1.5, 1.5, 1)),
+                labSize = 4.0,
+                col = c('grey30', 'forestgreen', 'royalblue', 'red2'),
+                colAlpha = 1,
+                legendLabels=c('Not Significant', 'Log2FC', 'P-Value', 'Log2FC & P-Value'),
+                legendPosition = 'right',
+                legendLabSize = 14,
+                drawConnectors = TRUE,
+                widthConnectors = 0.5,
+                colConnectors = 'grey30')
+
+```
+
+TD: List notable genes, highlight where they are found, what they do, etc
+
+## Automate this for pre-annotated cell types
+
+### Aggregate gene counts for each sample
+
+```{r}
+# TKTK: how do I filter for 0 gene counts of a seurat obj?
+# Filter genes with zero counts
+## Start loop, then filter
+annot_pseudobulk <- AggregateExpression(
+  se, # object name
+  group.by = c('Sample ID','Celltype')
+  )$RNA # desired assay
+# NOTE: Can add "return.seurat = TRUE" and assign this to a new Seurat obj
+```
+
+```{r}
+# We have to rearrange the data somewhat to get it into the DESeq object. colData is the sample-level metadata
+
+# Celltype variable is in question
+sample_info <- strsplit(colnames(annot_pseudobulk), "_")
+
+# TKTK: figure out consistent factor order
+# Prep variables
+lvls <- c("Normal", "Flu", "nCoV")
+ftr <- str_split(colnames(annot_pseudobulk), pattern = " ", simplify = TRUE)
+
+colData <- data.frame(
+  sample = colnames(annot_pseudobulk),
+  condition = factor(ftr[, 1], levels = lvls),
+  donor_id = sapply(sample_info, `[`, 1),
+  CT = sapply(sample_info, `[`, 2)
+)
+
+# Ensure unique rownames in colData for DESeqDataSet
+rownames(colData) <- colnames(annot_pseudobulk)
+
+colData <- colData %>%
+  left_join(se@meta.data %>%
+  dplyr::select(sample) %>%
+    distinct()
+) %>%
+  column_to_rownames(var = 'sample')
 ```
 
 ```{r}
-# plotMA()
-plotMA(res01)
+# Use lapply function to loop through Celltypes
+lapply(unique(colData$CT), function(cluster) {
```

Comment:
Can you add comments on what each line does within the lapply loop? I know that we've gone through all of it with the CD4s but I think it will be helpful for people to have!